### ISSUE TITLE
How to generate report with internal maven repository?

### ISSUE THREAD

**Issue by eroshenkoam** [jenkinsci/allure-plugin#11]
My environment compleltely isolated from the internet. But I have artifactory instance in internal network. How to point report builder to fetch artifacts from this repository url?

For now it fails with NPE

```
Allure Report Generation: started
Allure Report Generation: find directories by mask [target\allure-results]
Allure Report Generation: found allure result directories [D:\Jenkins\workspace\prj\target\allure-results]
Allure Report Generation: copy founded directories in directory [D:\Jenkins\workspace\prj\allure1008803587963066652.tmp\results]
Allure Report Generation: generate report from directory [D:\Jenkins\workspace\prj\allure1008803587963066652.tmp]
Allure Report Generation: proxy settings [active:'false', host:'null', port:'0', username:'null', password: '']
ERROR: Publisher ru.yandex.qatools.allure.jenkins.AllureReportPublisher aborted due to exception
java.io.IOException: ru.yandex.qatools.allure.report.AllureReportBuilderException: java.lang.NullPointerException
        at ru.yandex.qatools.allure.jenkins.utils.ReportGenerator.invoke(ReportGenerator.java:50)
        at ru.yandex.qatools.allure.jenkins.utils.ReportGenerator.invoke(ReportGenerator.java:25)
        at hudson.FilePath.act(FilePath.java:917)
        at hudson.FilePath.act(FilePath.java:895)
        at ru.yandex.qatools.allure.jenkins.AllureReportPublisher.generateReport(AllureReportPublisher.java:243)
        at ru.yandex.qatools.allure.jenkins.AllureReportPublisher.perform(AllureReportPublisher.java:144)
        at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
        at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:770)
        at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:734)
        at hudson.maven.MavenModuleSetBuild$MavenModuleSetBuildExecution.post2(MavenModuleSetBuild.java:1040)
        at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:683)
        at hudson.model.Run.execute(Run.java:1765)
        at hudson.maven.MavenModuleSetBuild.run(MavenModuleSetBuild.java:529)
        at hudson.model.ResourceController.execute(ResourceController.java:89)
        at hudson.model.Executor.run(Executor.java:240)
Caused by: ru.yandex.qatools.allure.report.AllureReportBuilderException: java.lang.NullPointerException
        at ru.yandex.qatools.allure.report.AllureReportBuilder.processResults(AllureReportBuilder.java:133)
        at ru.yandex.qatools.allure.jenkins.utils.ReportGenerator.invoke(ReportGenerator.java:47)
        ... 14 more
Caused by: java.lang.NullPointerException
        at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:361)
        at ru.yandex.qatools.clay.Aether.resolveWithTransitives(Aether.java:177)
        at ru.yandex.qatools.clay.Aether.resolve(Aether.java:131)
        at ru.yandex.qatools.clay.Aether.resolve(Aether.java:109)
        at ru.yandex.qatools.allure.report.AllureReportBuilder.processResults(AllureReportBuilder.java:127)
        ... 15 more
```

**Comment by kkucera** [comment]
I wish I could answer your question, but I get the exact same error when trying to generate reports with Jenkins and can't seem to figure out why it occurs. Do you have any insight into why this error occurs?

    Allure Report Generation: started
    Allure Report Generation: find directories by mask [**/build/allure-results]
    Allure Report Generation: found allure result directories [/var/lib/jenkins/workspace/SmokeTests-Firefox/build/allure-results]
    Allure Report Generation: copy founded directories in directory [/var/lib/jenkins/workspace/SmokeTests-Firefox/allure8288703328821124072.tmp/results]
    Allure Report Generation: generate report from directory [/var/lib/jenkins/workspace/SmokeTests-Firefox/allure8288703328821124072.tmp]
    Allure Report Generation: proxy settings [active:'false', host:'null', port:'0', username:'null', password: '']
    ERROR: Publisher 'Allure Report Generation' aborted due to exception: 
    java.io.IOException: ru.yandex.qatools.allure.report.AllureReportBuilderException: java.lang.NullPointerException
	at ru.yandex.qatools.allure.jenkins.utils.ReportGenerator.invoke(ReportGenerator.java:50)
	at ru.yandex.qatools.allure.jenkins.utils.ReportGenerator.invoke(ReportGenerator.java:25)
	at hudson.FilePath.act(FilePath.java:991)
	at hudson.FilePath.act(FilePath.java:969)
	at ru.yandex.qatools.allure.jenkins.AllureReportPublisher.generateReport(AllureReportPublisher.java:267)
	at ru.yandex.qatools.allure.jenkins.AllureReportPublisher.perform(AllureReportPublisher.java:146)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:779)
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:726)
	at hudson.model.Build$BuildExecution.post2(Build.java:185)
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:671)
	at hudson.model.Run.execute(Run.java:1766)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:381)
    Caused by: ru.yandex.qatools.allure.report.AllureReportBuilderException: java.lang.NullPointerException
	at ru.yandex.qatools.allure.report.AllureReportBuilder.processResults(AllureReportBuilder.java:133)
	at ru.yandex.qatools.allure.jenkins.utils.ReportGenerator.invoke(ReportGenerator.java:47)
	... 14 more
    Caused by: java.lang.NullPointerException
	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:361)
	at ru.yandex.qatools.clay.Aether.resolveWithTransitives(Aether.java:177)
	at ru.yandex.qatools.clay.Aether.resolve(Aether.java:131)
	at ru.yandex.qatools.clay.Aether.resolve(Aether.java:109)
	at ru.yandex.qatools.allure.report.AllureReportBuilder.processResults(AllureReportBuilder.java:127)

I also get the NPE if I try to generate the reports from the command line tool. This only happens if I sudo my apache user. Which is the same user jenkins runs under.

sudo -u apache allure --debug generate /webpt/test-automation/automated-regression-smoke-tests/build/allure-results/ -o /tmp -v 1.4.16

    Generating report for Allure version [1.4.16]
    Processing result pattern [/webpt/test-automation/automated-regression-smoke-tests/build/allure-results/]
    Found results directory [/webpt/test-automation/automated-regression-smoke-tests/build/allure-results]
    java.lang.NullPointerException
    ru.yandex.qatools.allure.report.AllureReportBuilderException: java.lang.NullPointerException

If I generate the reports using my regular account (not sudo -u apache) they work fine. 

It sounds like you believe you were getting the error because of a network connectivity issue? Which would mean that my apache user has some network restriction? I'm just guessing because the error messaging doesn't really point me in a direction. Let me know if you have any insight, thanks.

**Comment by eroshenkoam** [comment]
fixed in 2.8
